core/src/main/scala/org/apache/spark/util/ClosureCleaner.scala

```diff
@@ -274,7 +265,7 @@ private[spark] object ClosureCleaner extends Logging {
       // required fields from the original object. We need the parent here because the Java
       // language specification requires the first constructor parameter of any closure to be
       // its enclosing object.
-      val clone = instantiateClass(cls, parent, inInterpreter)
+      val clone = instantiateClass(cls, parent, Utils.isInInterpreter)
```

Comment:
if we're getting it from `Utils` we don't even need to pass it into this argument. You can just replace its usage down there inside `instantiateClass` itself.